libs/data-mapper-v2/src/ui/DMReactFlow.tsx

```diff
@@ -0,0 +1,233 @@
+import type { AppDispatch, RootState } from '../core/state/Store';
+import { useEffect, useMemo, useRef, useCallback, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import type { Connection, Node, Edge, ConnectionLineComponent, NodeDragHandler } from 'reactflow';
+import ReactFlow, { addEdge, useReactFlow } from 'reactflow';
+import { reactFlowStyle, useStaticStyles, useStyles } from './styles';
+import SchemaNode from '../components/common/reactflow/SchemaNode';
+import ConnectionLine from '../components/common/reactflow/ConnectionLine';
+import ConnectedEdge from '../components/common/reactflow/ConnectedEdge';
+import type { ConnectionAction } from '../core/state/DataMapSlice';
+import { makeConnection, updateReactFlowEdges, updateReactFlowNodes } from '../core/state/DataMapSlice';
+import { FunctionNode } from '../components/common/reactflow/FunctionNode';
+import { useDrop } from 'react-dnd';
+import useResizeObserver from 'use-resize-observer';
+
+interface DMReactFlowProps {
+  setIsMapStateDirty?: (isMapStateDirty: boolean) => void;
+  updateCanvasBoundsParent: (bounds: DOMRect | undefined) => void;
+}
+
+export const DMReactFlow = ({ setIsMapStateDirty, updateCanvasBoundsParent }: DMReactFlowProps) => {
+  useStaticStyles();
+  const styles = useStyles();
+  const reactFlowInstance = useReactFlow();
+  const ref = useRef<HTMLDivElement>(null);
+  const dispatch = useDispatch<AppDispatch>();
+  const [allNodes, setAllNodes] = useState<Node[]>([]);
+
+  const { width = -1, height = -1 } = useResizeObserver<HTMLDivElement>({
+    ref,
+  });
+
+  useEffect(() => {
+    if (ref?.current) {
+      const bounds = ref.current.getBoundingClientRect();
+      bounds.width = width;
+      bounds.height = height;
+      updateCanvasBoundsParent(bounds);
+    }
+  }, [ref, updateCanvasBoundsParent, width, height]);
```

Comment:
This logic has changed in the latest main. We no longer maintain the canvasBounds. But maybe I can take it up in my PR.